src/SaveObject.ts

```diff
@@ -719,6 +721,39 @@ Error: ${e}`);
     Object.assign(freshSaveData.stats, stats);
     loadGo(JSON.stringify(freshSaveData));
   }
+  if (ver <= 38) {
+    // All whitespace except for spaces was allowed in filenames
+    let found = false;
+    for (const server of GetAllServers()) {
+      for (const script of server.scripts.values()) {
+        if (!/\s/.test(script.filename)) continue;
+        removeWhitespace(server.hostname, script, server.scripts);
+        found = true;
+      }
+      for (const textFile of server.textFiles.values()) {
+        if (!/\s/.test(textFile.filename)) continue;
+        removeWhitespace(server.hostname, textFile, server.textFiles);
+        found = true;
+      }
+    }
+    if (found) Terminal.error("Filenames with whitespace found and corrected, see console for details.");
+  }
+}
+
+function removeWhitespace(hostname: ServerName, file: ContentFile, files: ContentFileMap): void {
+  let filename = file.filename.replace(/\s+/g, "-") as ContentFilePath;
+  // avoid filename conflicts
+  if (files.has(filename)) {
+    const [path, ext] = filename.split(".");
```

Comment:
Files can have . in them before the extension, this doesn't reliably provide you with the extension

Reply:
Yeah that should work, the last . and anything after it is always the extension